src/qtt/algorithms/functions.py

```diff
@@ -215,13 +215,49 @@ def _double_gaussian(x, A_dn, A_up, sigma_dn, sigma_up, mean_dn, mean_up):
     result_dict = {'parameters initial guess': initial_params,
                    'separation': separation, 'split': weigthed_distance_split}
     result_dict['parameters'] = par_fit
+    result_dict['reduced_chi_squared'] = result.redchi
+
     result_dict['left'] = np.take(par_fit, [4, 2, 0])
     result_dict['right'] = np.take(par_fit, [5, 3, 1])
     result_dict['type'] = 'fitted double gaussian'
 
     return par_fit, result_dict
 
 
+def refit_double_gaussian(result_dict, x_data, y_data, gaussian_amplitude_ratio_threshold=8):
```

Comment:
Can this method be called without first calling fit_double_gaussian? If not, why not call fit_double_gaussian from within this method (the refit... part of the name is not accurate then of course).

Reply:
I cannot be called without first calling `fit_double_gaussian` (unless someone prepares data similar to the output of that method). However, if build this into the method, then a user has less freedom in calling the method. The current approach seems the most modular to me